pkg/datapath/linux/probes/probes.go

```diff
@@ -201,85 +208,96 @@ func (p *ProbeManager) SystemKernelHz() (int, error) {
-	kernelParams := make(map[KernelParam]bool)
+	kernelParams := make(map[KernelParam]kernelOption)
 
-	kernelParams["CONFIG_BPF"] = config.ConfigBpf.Enabled()
-	kernelParams["CONFIG_BPF_SYSCALL"] = config.ConfigBpfSyscall.Enabled()
-	kernelParams["CONFIG_NET_SCH_INGRESS"] = config.ConfigNetSchIngress.Enabled()
-	kernelParams["CONFIG_NET_CLS_BPF"] = config.ConfigNetClsBpf.Enabled()
-	kernelParams["CONFIG_NET_CLS_ACT"] = config.ConfigNetClsAct.Enabled()
-	kernelParams["CONFIG_BPF_JIT"] = config.ConfigBpfJit.Enabled()
-	kernelParams["CONFIG_HAVE_EBPF_JIT"] = config.ConfigHaveEbpfJit.Enabled()
+	kernelParams["CONFIG_BPF"] = kernelOption{
+		Enabled:     config.ConfigBpf.Enabled(),
+		Description: "",
+		CanBeModule: false,
+	}
+	kernelParams["CONFIG_BPF_SYSCALL"] = kernelOption{
+		Enabled:     config.ConfigBpfSyscall.Enabled(),
+		Description: "",
+		CanBeModule: false,
+	}
+	kernelParams["CONFIG_NET_SCH_INGRESS"] = kernelOption{
+		Enabled:     config.ConfigNetSchIngress.Enabled(),
+		Description: "",
+		CanBeModule: true,
+	}
+	kernelParams["CONFIG_NET_CLS_BPF"] = kernelOption{
+		Enabled:     config.ConfigNetClsBpf.Enabled(),
+		Description: "",
+		CanBeModule: true,
+	}
+	kernelParams["CONFIG_NET_CLS_ACT"] = kernelOption{
+		Enabled:     config.ConfigNetClsAct.Enabled(),
+		Description: "",
+		CanBeModule: false,
+	}
+	kernelParams["CONFIG_BPF_JIT"] = kernelOption{
+		Enabled:     config.ConfigBpfJit.Enabled(),
+		Description: "",
+		CanBeModule: false,
+	}
+	kernelParams["CONFIG_HAVE_EBPF_JIT"] = kernelOption{
+		Enabled:     config.ConfigHaveEbpfJit.Enabled(),
+		Description: "",
+		CanBeModule: false,
+	}
 
 	return kernelParams
 }
 
 // GetOptionalConfig performs a check of *optional* kernel configuration options. It
 // returns a map indicating which optional/non-mandatory kernel paramaters are enabled.
 // GetOptionalConfig is being used by CLI "cilium kernel-check".
-func (p *ProbeManager) GetOptionalConfig() map[KernelParam]bool {
+func (p *ProbeManager) GetOptionalConfig() map[KernelParam]kernelOption {
 	config := p.features.SystemConfig
-	kernelParams := make(map[KernelParam]bool)
+	kernelParams := make(map[KernelParam]kernelOption)
 
-	kernelParams["CONFIG_CGROUP_BPF"] = config.ConfigCgroupBpf.Enabled()
-	kernelParams["CONFIG_LWTUNNEL_BPF"] = config.ConfigLwtunnelBpf.Enabled()
-	kernelParams["CONFIG_BPF_EVENTS"] = config.ConfigBpfEvents.Enabled()
+	kernelParams["CONFIG_CGROUP_BPF"] = kernelOption{
+		Enabled:     config.ConfigCgroupBpf.Enabled(),
+		Description: "Host Reachable Services and Sockmap optimization",
+		CanBeModule: false,
+	}
+	kernelParams["CONFIG_LWTUNNEL_BPF"] = kernelOption{
+		Enabled:     config.ConfigLwtunnelBpf.Enabled(),
+		Description: "Leightweight Tunnel Hook for IP-in-IP encapsulation",
+		CanBeModule: false,
+	}
+	kernelParams["CONFIG_BPF_EVENTS"] = kernelOption{
+		Enabled:     config.ConfigBpfEvents.Enabled(),
+		Description: "Enables Visibility and Introspection into Datapath",
```

Comment:
```suggestion
		Description: "Visibility and introspection into datapath",
```
Sorry my earlier comment was unclear, I thought title case for feature names, not necessarily for the whole description.